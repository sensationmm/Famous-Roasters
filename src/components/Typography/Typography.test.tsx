import { render, waitFor } from '@testing-library/react'
import React from 'react'

import { Typography, TypographySize, TypographyType } from '.'

describe('Typography component', () => {
  describe('Heading', () => {
    it('Renders correctly for size large', async () => {
      const { container } = render(
        <Typography as="h1" type={TypographyType.Heading} size={TypographySize.Large}>
          Text
        </Typography>,
      )
      await waitFor(() => new Promise((res) => setTimeout(res, 0)))
      expect(container).toMatchSnapshot()
    })

    it('Renders correctly for size base', async () => {
      const { container } = render(
        <Typography as="h1" type={TypographyType.Heading}>
          Text
        </Typography>,
      )
      await waitFor(() => new Promise((res) => setTimeout(res, 0)))
      expect(container).toMatchSnapshot()
    })

    it('Renders correctly for size small', async () => {
      const { container } = render(
        <Typography as="h1" type={TypographyType.Heading} size={TypographySize.Small}>
          Text
        </Typography>,
      )
      await waitFor(() => new Promise((res) => setTimeout(res, 0)))
      expect(container).toMatchSnapshot()
    })

    it('Renders correctly for size tiny', async () => {
      const { container } = render(
        <Typography as="h1" type={TypographyType.Heading} size={TypographySize.Tiny}>
          Text
        </Typography>,
      )
      await waitFor(() => new Promise((res) => setTimeout(res, 0)))
      expect(container).toMatchSnapshot()
    })
  })

  describe('Label', () => {
    it('Renders correctly for size large', async () => {
      const { container } = render(
        <Typography as="span" type={TypographyType.Label} size={TypographySize.Large}>
          Text
        </Typography>,
      )
      await waitFor(() => new Promise((res) => setTimeout(res, 0)))
      expect(container).toMatchSnapshot()
    })

    it('Renders correctly for size base', async () => {
      const { container } = render(<Typography type={TypographyType.Label}>Text</Typography>)
      await waitFor(() => new Promise((res) => setTimeout(res, 0)))
      expect(container).toMatchSnapshot()
    })

    it('Renders correctly for size small', async () => {
      const { container } = render(
        <Typography as="span" type={TypographyType.Label} size={TypographySize.Small}>
          Text
        </Typography>,
      )
      await waitFor(() => new Promise((res) => setTimeout(res, 0)))
      expect(container).toMatchSnapshot()
    })

    it('Renders correctly for size tiny', async () => {
      const { container } = render(
        <Typography as="span" type={TypographyType.Label} size={TypographySize.Tiny}>
          Text
        </Typography>,
      )
      await waitFor(() => new Promise((res) => setTimeout(res, 0)))
      expect(container).toMatchSnapshot()
    })
  })

  describe('Paragraph', () => {
    it('Renders correctly for size large', async () => {
      const { container } = render(
        <Typography as="p" type={TypographyType.Paragraph} size={TypographySize.Large}>
          Text
        </Typography>,
      )
      await waitFor(() => new Promise((res) => setTimeout(res, 0)))
      expect(container).toMatchSnapshot()
    })

    it('Renders correctly for size base', async () => {
      const { container } = render(<Typography as="p">Text</Typography>)
      await waitFor(() => new Promise((res) => setTimeout(res, 0)))
      expect(container).toMatchSnapshot()
    })

    it('Renders correctly for size small', async () => {
      const { container } = render(
        <Typography as="p" type={TypographyType.Paragraph} size={TypographySize.Small}>
          Text
        </Typography>,
      )
      await waitFor(() => new Promise((res) => setTimeout(res, 0)))
      expect(container).toMatchSnapshot()
    })

    it('Renders correctly for size tiny', async () => {
      const { container } = render(
        <Typography as="p" type={TypographyType.Paragraph} size={TypographySize.Tiny}>
          Text
        </Typography>,
      )
      await waitFor(() => new Promise((res) => setTimeout(res, 0)))
      expect(container).toMatchSnapshot()
    })
  })
})
