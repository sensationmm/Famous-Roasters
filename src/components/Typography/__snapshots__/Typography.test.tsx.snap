// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography component Heading Renders correctly for size base 1`] = `
<div>
  <h1
    class="text-3xl leading-8 font-semibold tracking-tight"
  >
    Text
  </h1>
</div>
`;

exports[`Typography component Heading Renders correctly for size large 1`] = `
<div>
  <h1
    class="text-5xl leading-none font-normal"
  >
    Text
  </h1>
</div>
`;

exports[`Typography component Heading Renders correctly for size small 1`] = `
<div>
  <h1
    class="text-2xl leading-8 font-semibold"
  >
    Text
  </h1>
</div>
`;

exports[`Typography component Heading Renders correctly for size tiny 1`] = `
<div>
  <h1
    class="text-xl leading-7 font-semibold"
  >
    Text
  </h1>
</div>
`;

exports[`Typography component Label Renders correctly for size base 1`] = `
<div>
  <span
    class="text-base leading-6 font-semibold"
  >
    Text
  </span>
</div>
`;

exports[`Typography component Label Renders correctly for size large 1`] = `
<div>
  <span
    class="text-lg leading-7 font-semibold"
  >
    Text
  </span>
</div>
`;

exports[`Typography component Label Renders correctly for size small 1`] = `
<div>
  <span
    class="text-sm leading-5 font-semibold"
  >
    Text
  </span>
</div>
`;

exports[`Typography component Label Renders correctly for size tiny 1`] = `
<div>
  <span
    class="text-xs leading-4 font-semibold"
  >
    Text
  </span>
</div>
`;

exports[`Typography component Paragraph Renders correctly for size base 1`] = `
<div>
  <p
    class="text-base leading-6 font-normal"
  >
    Text
  </p>
</div>
`;

exports[`Typography component Paragraph Renders correctly for size large 1`] = `
<div>
  <p
    class="text-lg leading-7 font-normal"
  >
    Text
  </p>
</div>
`;

exports[`Typography component Paragraph Renders correctly for size small 1`] = `
<div>
  <p
    class="text-sm leading-5 font-normal"
  >
    Text
  </p>
</div>
`;

exports[`Typography component Paragraph Renders correctly for size tiny 1`] = `
<div>
  <p
    class="text-xs leading-4 font-normal"
  >
    Text
  </p>
</div>
`;
